### PR TITLE
Handle empty body on fetch response

### PR DESCRIPTION
**Pull request by kevkehoe**
Currently the fetch will attempt to call json.parse on the body which throws if the body is empty. Don't think the check for a 204 is enough because from the HTTP spec:

*4.3 Message Body*

*... All 1xx (informational), 204 (no content), and 304 (not modified) responses MUST NOT include a message-body. All other responses do include a message-body, although* **it MAY be of zero length.**

Also I'm using it in a real world scenario where I'm returning a 401 with no body content. I've replaced the .json call with an evaluation of the response text and the a json.parse of an empty object if the text length is zero. A try parse on the text would probably have been better but i don't know a decent way of doing it.

I didn't dig too deep and there may well be a better approach to this or the possibility to move it into the .json function - also the tests might be pony.

:ghost:

### CODE REVIEW

**Comment by RobinBressan** [comment]
What about only checking for the status code? This way we keep consistant with the http spec.

**Comment by kevkehoe** [comment]
I'm not sure it's possible just from a check of the status code though, unless you mean ensure there's **no body** for a *1xx*, *204* or *304* status and always include the body (even if it's empty) if it's any other status... That seems to me the only way to handle the HTTP spec which states the message body **MAY be of zero length**. In this instance that would mean extending the check for the status codes and updating the response.json() function to return an empty value.

Basically, the only issue with how it works now is the fact the json function errors if the value is null or empty.

File: test/src/http/fetchSpec.js [hunk 1]
```diff
@@ -112,7 +114,7 @@ describe('Fetch HTTP Backend', () => {
         })
         .then((_response) => {
             expect(_response).to.deep.equal({
-                data: null,
+                data: {},
```
Comment (RobinBressan) [inline test/src/http/fetchSpec.js]: I think we should always returns a `null` body if it is empty. Can you fix this and I'll merge you next ?

**Comment by RobinBressan** [comment]
superseded by #70 thanks for pointing this issue!
